pkg/apis/v1/nodeclaim.go

```diff
@@ -55,6 +52,14 @@ type NodeClaimSpec struct {
 	NodeClassRef *NodeClassReference `json:"nodeClassRef"`
 }
 
+// NodeClaimSpec describes the desired state of the NodeClaim
+type NodeClaimSpec struct {
+	NodeClaimTemplateSpec `json:",inline"`
```

Comment:
This naming feels odd to me. What if we named this NodeClaimTemplate?
```suggestion
// NodeClaimSpec describes the desired state of the NodeClaim
type NodeClaimTemplate struct {
	NodeClaimTemplateSpec `json:",inline"`
```

Reply:
Update the name of NodeClaimTemplateSpec to be NodeClaimCoreSpec